File: CS-Examples/06_Text/SetSuperscriptAndSubscript.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace SetSuperscriptAndSubscript
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class.
			Document document = new Document();

			// Add a section to the document.
			Section section = document.AddSection();

			// Add a paragraph to the section.
			Paragraph paragraph = section.AddParagraph();

			// Append the text "E = mc" to the paragraph.
			paragraph.AppendText("E = mc");

			// Append the text "2" as a superscript to the paragraph.
			TextRange range1 = paragraph.AppendText("2");
			range1.CharacterFormat.SubSuperScript = SubSuperScript.SuperScript;

			// Insert a line break in the paragraph.
			paragraph.AppendBreak(BreakType.LineBreak);

			// Append the text "F" to the paragraph.
			paragraph.AppendText("F");

			// Append the text "n" as a subscript to the paragraph.
			TextRange range2 = paragraph.AppendText("n");
			range2.CharacterFormat.SubSuperScript = SubSuperScript.SubScript;

			// Append the text " = Fn-1 + Fn-2" with specific subscripts to the paragraph.
			paragraph.AppendText(" = F");
			paragraph.AppendText("n-1").CharacterFormat.SubSuperScript = SubSuperScript.SubScript;
			paragraph.AppendText(" + F");
			paragraph.AppendText("n-2").CharacterFormat.SubSuperScript = SubSuperScript.SubScript;

			// Set the font size to 36 for all TextRange items in the paragraph.
			foreach (var i in paragraph.Items)
			{
				if (i is TextRange)
				{
					(i as TextRange).CharacterFormat.FontSize = 36;
				}
			}

			// Specify the output file name.
			string output = "SetSuperscriptAndSubscript.docx";

			// Save the document to a file with the specified output file name and format (Docx).
			document.SaveToFile(output, FileFormat.Docx);

			// Clean up resources used by the document.
			document.Dispose();

            //Launching the file
            WordDocViewer(output);

        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/06_Text/SetSuperscriptAndSubscript.vb
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace SetSuperscriptAndSubscript
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object.
			Dim document As New Document()

			' Add a section to the document.
			Dim section As Section = document.AddSection()

			' Add a paragraph to the section.
			Dim paragraph As Paragraph = section.AddParagraph()
			paragraph.AppendText("E = mc")

			' Append the superscript "2" to the paragraph and store it in range1.
			Dim range1 As TextRange = paragraph.AppendText("2")
			range1.CharacterFormat.SubSuperScript = SubSuperScript.SuperScript

			' Insert a line break in the paragraph.
			paragraph.AppendBreak(BreakType.LineBreak)

			' Append the text "F" to the paragraph and store it in range2.
			paragraph.AppendText("F")
			Dim range2 As TextRange = paragraph.AppendText("n")
			range2.CharacterFormat.SubSuperScript = SubSuperScript.SubScript

			' Append more text to the paragraph with subscripts.
			paragraph.AppendText(" = F")
			paragraph.AppendText("n-1").CharacterFormat.SubSuperScript = SubSuperScript.SubScript
			paragraph.AppendText(" + F")
			paragraph.AppendText("n-2").CharacterFormat.SubSuperScript = SubSuperScript.SubScript

			' Iterate through each item in the paragraph and set the font size to 36 for TextRange objects.
			For Each i In paragraph.Items
				If TypeOf i Is TextRange Then
					TryCast(i, TextRange).CharacterFormat.FontSize = 36
				End If
			Next i

			' Save the document to a file named "SetSuperscriptAndSubscript.docx".
			Dim output As String = "SetSuperscriptAndSubscript.docx"
			document.SaveToFile(output, FileFormat.Docx)

			' Dispose of the document object to free up resources.
			document.Dispose()

			'Launching the file
			WordDocViewer(output)

		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
